addons/website_event_booth_sale/controllers/event_booth.py

```diff
@@ -20,20 +21,31 @@ def event_booth_main(self, event):
     @http.route()
     def event_booth_registration_confirm(self, event, booth_category_id, event_booth_ids, **kwargs):
         booths = self._get_requested_booths(event, event_booth_ids)
+        if not booths:
+            return json.dumps({'error': 'boothError'})
+
         booth_category = request.env['event.booth.category'].sudo().browse(int(booth_category_id))
+        if not booth_category.exists():
+            return json.dumps({'error': 'boothCategoryError'})
+
+        booth_values = self._prepare_booth_registration_values(event, kwargs)
         order = request.website.sale_get_order(force_create=1)
         order._cart_update(
             product_id=booth_category.product_id.id,
             set_qty=1,
             event_booth_pending_ids=booths.ids,
-            registration_values=self._prepare_booth_registration_values(event, kwargs),
+            registration_values=booth_values,
         )
         if order.amount_total:
-            return request.redirect('/shop/checkout')
+            return json.dumps({'redirect': '/shop/checkout'})
         elif order:
             order.action_confirm()
             request.website.sale_reset()
 
-        return request.redirect(('/event/%s/booth/success?' % event.id) + werkzeug.urls.url_encode({
-            'booths': ','.join([str(id) for id in booths.ids]),
-        }))
+        return json.dumps({
+            'success': True,
+            'event_name': event.name,
+            'contact': {
+                key.partition('contact_')[2]: val for key, val in booth_values.items() if
+            key.startswith('contact_')},
+        })
```

Comment:
Mmh.

Not a fan of the fact that we are not calling super here and building the same kind of response twice.
This makes maintaining the code a bit more complicated.

Maybe we could:

```py
if order.amount_total:
    # bypass super method logic to go through checkout process
    return request.redirect('/shop/checkout')
else:
    result = # call super
    request.website.sale_reset()
    return result
```

What do you think?